rhp/v2/contracts.go

```diff
@@ -95,7 +95,7 @@ func ContractRenewalCost(fc types.FileContract, contractFee types.Currency) type
 }
 
 // PrepareContractRenewal constructs a contract renewal transaction.
-func PrepareContractRenewal(currentRevision types.FileContractRevision, renterKey PrivateKey, hostKey PublicKey, renterPayout, hostCollateral types.Currency, endHeight uint64, host HostSettings, refundAddr types.UnlockHash) types.FileContract {
+func PrepareContractRenewal(currentRevision types.FileContractRevision, renterKey PrivateKey, hostKey PublicKey, renterPayout types.Currency, endHeight uint64, host HostSettings, refundAddr types.UnlockHash) types.FileContract {
```

Comment:
huh, so `hostCollateral` was unused. Is that correct? *Should* it be used somehow, or can `hostCollateral` always be calculated from the other parameters?

Reply:
Yes, it was unused. The reason being that we infer the collateral from the renter payout by computing a cost-per-byte and then seeing how much additional data we think a user would upload to the contract.
In theory we can do the same for forming contracts but it seems like we don't and instead rely on the caller supplying the collateral.

Reply:
Ideally the worker shouldn't be estimating anything. It's just a dumb conduit. The autopilot should be the one making any decisions around how much data to store and how much funding/collateral a contract should have.